cmd/fs-v1.go

```diff
@@ -1079,7 +1079,10 @@ func (fs *fsObjects) ListBucketsHeal() ([]BucketInfo, error) {
 
 // SetBucketPolicy sets policy on bucket
 func (fs *fsObjects) SetBucketPolicy(bucket string, policy policy.BucketAccessPolicy) error {
-	return persistAndNotifyBucketPolicyChange(bucket, false, policy, fs)
+	if reflect.DeepEqual(policy, emptyBucketPolicy) {
+		return errInvalidArgument
+	}
+	return writeBucketPolicy(bucket, fs, policy)
```

Comment:
@harshavardhana , the S3PeersUpdateBucketPolicy(bucket) was also refreshing the policy cache - both SetBucketPolicy and DeleteBucketPolicy  will need to do that.